### PR TITLE
feat: button for refreshing the estimated calls list

### PR DESCRIPTION
**Pull request by mortennordseth**
This introduces a button to refresh the estimated calls list.

<details>
<summary>Screenshots</summary>

image

image

</details>

### TODO
- [x] Add correct icon (waiting on https://github.com/AtB-AS/design-system/pull/145)

Closes https://github.com/AtB-AS/kundevendt/issues/9149

### CODE REVIEW

**Comment by mortennordseth** [comment]
FYI @mikaelbr, I misread the Figma sketches when I started working on the refresh button for the estimated calls list. As I'd already started working on the functionality needed for the nearest-stop-places page I chose to include it in this PR, making this PR cover both refresh buttons. This is still WIP and I'll continue this work after I've looked at the other open PRs. I just wanted to mention it since you've been working on the nearest-stop-places page

**Comment by mikaelbr** [comment]
I don't think refresh on stop page makes sense. So I didn't include it in my original page. There is nothing to refresh, so I think it's just an error in the design

**Comment by mortennordseth** [comment]
@mikaelbr or @adriansberg , do you see a better way to handle the icon color when hovering over the link? With the current solution the text changes color before the icon and makes it a bit weird. 

(also ignore the icon type, need to merge this before I can add the correct one: https://github.com/AtB-AS/design-system/pull/145)

https://github.com/AtB-AS/planner-web/assets/43166974/f814660a-a5ab-45f5-abf8-bc56b0859f2e

**Comment by adriansberg** [comment]
> @mikaelbr or @adriansberg , do you see a better way to handle the icon color when hovering over the link? With the current solution the text changes color before the icon and makes it a bit weird.
> 
> (also ignore the icon type, need to merge this before I can add the correct one: [AtB-AS/design-system#145](https://github.com/AtB-AS/design-system/pull/145))
> 
>  Skjermopptak.2023-10-16.kl.13.34.02.mov

If I understood it correctly, https://github.com/AtB-AS/kundevendt/issues/9140 sort of fixes this by enabling the icons to be manipulated by css, thus being able to add the wanted transition of color to match the transition of the text.

I can't think of any solution to this with the current setup for icons.

A work-around might be to remove the color transition of the text so that it's less weird for now, but not sure how it will be with this button (and buttons with icon) being different than others.

**Comment by mortennordseth** [comment]
> A work-around might be to remove the color transition of the text so that it's less weird for now

This makes it less weird for sure. Not the best solution, but as you say, the other issue (https://github.com/AtB-AS/kundevendt/issues/9140) might fix this.

File: src/components/icon/mono-icon.tsx [hunk 1]
```diff
@@ -19,18 +20,31 @@ export type MonoIconProps = Omit<JSX.IntrinsicElements['img'], 'src'> & {
    * @default none
    */
   overrideMode?: 'none' | 'dark' | 'light';
+  /**
+   * Used to override light- or dark mode icon color when MonoIcon is used with an interactive component
+   */
+  interactiveColor?: keyof Theme['interactive'];
```
Comment (adriansberg) [inline src/components/icon/mono-icon.tsx:26]: Are `interactiveColor` and `interactiveState` both going to be present at the same time?

File: src/styles/globals.css [hunk 2]
```diff
@@ -99,8 +99,7 @@ textarea {
   --contentPadding: var(--spacings-xLarge);
 }
 
-a,
-a:visited {
```
Comment (adriansberg) [inline src/styles/globals.css:103]: Why was this removed?

**Comment by mikaelbr** [comment]
Icon color here is one of the reasons why I want https://github.com/AtB-AS/kundevendt/issues/9140 as you've said. I think this technically could be solved for mono icons using CSS filters, but I think it's fine as is for now. But I'd want to verify that the contrast is actually correct there though. Even though it uses theme colors, there might be a contrast issue in the theme design.

File: src/components/icon/mono-icon.tsx (hunk 1 above)
Comment (mortennordseth) [reply src/components/icon/mono-icon.tsx:26]: It can be. There is no need for the `interactiveState` if you don't use the `interactiveColor`, but you can use the `interactiveColor` without the `interactiveState`. Most likely you would use both at the same time to handle hover/disabled colors etc.

File: src/styles/globals.css (hunk 2 above)
Comment (mortennordseth) [reply src/styles/globals.css:103]: The reason for this was that this overrides the interactive color on the refresh button. From what I can see, this is necessary for the links in the footer, but that is already covered here: https://github.com/AtB-AS/planner-web/blob/main/src/layouts/shared/footer.module.css#L46. 

Let me know if you see any reason why we need this 👍

**Comment by mortennordseth** [comment]
AtBs interactive 1 color for hover passes WCAG AA for normal text.

File: src/styles/globals.css (hunk 2 above)
Comment (adriansberg) [reply src/styles/globals.css:103]: I guess it's wanted in cases we have links in other places than e.g. the footer where it's covered. But not too big of a deal.

**Review by adriansberg — APPROVED** [review]
